src/hb-ot-standard-glyph-order-static.hh

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright © 2021  Google, Inc.
+ *
+ *  This is part of HarfBuzz, a text shaping library.
+ *
+ * Permission is hereby granted, without written agreement and without
+ * license or royalty fees, to use, copy, modify, and distribute this
+ * software and its documentation for any purpose, provided that the
+ * above copyright notice and the following two paragraphs appear in
+ * all copies of this software.
+ *
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE TO ANY PARTY FOR
+ * DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES
+ * ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN
+ * IF THE COPYRIGHT HOLDER HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * THE COPYRIGHT HOLDER SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING,
+ * BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE.  THE SOFTWARE PROVIDED HEREUNDER IS
+ * ON AN "AS IS" BASIS, AND THE COPYRIGHT HOLDER HAS NO OBLIGATION TO
+ * PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+ *
+ */
+#ifndef HB_OT_STANDARD_GLYPH_ORDER_STATIC_HH
+#define HB_OT_STANDARD_GLYPH_ORDER_STATIC_HH
+
+/* Following table is generated based on standardGlyphOrder.py
+ * in FontTools
+ */
+
+struct hb_ot_standard_glyph_order_map_t
+{
+  int cmp (const hb_bytes_t key) const
+  {
+    const char *g = glyph;
+    for (unsigned i = 0; i < key.length; i++)
+    {
+      char k = key.arrayZ[i];
+      if (*g == '\0' || k > *g) return 1;
+      if (k < *g) return -1;
+      g++;
+    }
+    if (*g == '\0') return 0;
+    return -1;
```

Comment:
Why are you not using `strcmp`?!

Reply:
oh, I found arrayZ in hb_bytes_t doesn't include a trailing '\0'. I will change to strcmp

Reply:
done